pkg/runtime/runtime.go

```diff
@@ -913,10 +867,13 @@ func (a *DaprRuntime) subscribeTopic(parentCtx context.Context, name string, top
 		policyRunner := resiliency.NewRunner[any](ctx, policyDef)
 		_, err = policyRunner(func(ctx context.Context) (any, error) {
 			var pErr error
-			if a.runtimeConfig.ApplicationProtocol.IsHTTP() {
+			switch a.runtimeConfig.ApplicationProtocol {
```

Comment:
Please revert this and use `IsHTTP`, because we now have `https` and `h2c` too, and for gRPC (`!IsHTTP`) we have `grpcs`